var/spack/repos/builtin/packages/ncurses/package.py

```diff
@@ -31,6 +31,9 @@ class Ncurses(AutotoolsPackage, GNUMirrorPackage):
     variant('termlib', default=True,
             description='Enables termlib features. This is an extra '
                         'lib and optional internal dependency.')
+    variant('abi5', default=False,
+            description='Use application binary interface (ABI) for '
+                        'ncurses version 5')
```

Comment:
The Spack HDF5 package defines a variant analogous to this:
```python
    variant('api', default='none', description='choose api compatibility', values=('none', 'v114', 'v112', 'v110', 'v18', 'v16'), multi=False)
```

I think a valued variant here would ultimately be a safer and would better match the other `api` variants in the codebase. Values `'none', '5', '6'`? Although I wish HDF5 had labeled theirs 'default' or 'native' or something rather than "no API" which is senseless...